libraries/AP_HAL_ChibiOS/SPIDevice.cpp

```diff
@@ -383,6 +392,25 @@ void SPIBus::start_peripheral(void)
     spi_started = true;
 }
 
+/*
+    check if the SPI configuration has changed since the last time
+*/
+bool SPIBus::config_changed(void)
+{
+    if (
+#if defined(STM32H7)
+        last_spicfg.cfg1 == spicfg.cfg1 &&
+        last_spicfg.cfg2 == spicfg.cfg2
+#else
+        last_spicfg.cr1 == spicfg.cr1 &&
+        last_spicfg.cr2 == spicfg.cr2
```

Comment:
I feel like you might need to check the DMA lock state here as well. If the lock is shared (and therefore the channel shared), the periperal may need to be shutdown to allow another to use the DMA channel.

Reply:
@andyp1per we only get to this line near the end of acquire bus which must have been successful.